packages/cli/src/commands/setup/deploy/deploy.js

```diff
@@ -83,15 +109,24 @@ export const handler = async ({ provider, force }) => {
             })
           ),
       },
-      providerData?.apiPackages?.length && {
+      providerData?.prismaDataSourceCheck && {
+        title: 'Checking Prisma data source provider...',
+        task: () => {
+          const fileData = providerData.prismaDataSourceCheck(database)
+          let files = {}
+          files[fileData.path] = fileData.content
+          return writeFilesTask(files, { overwriteExisting: force })
+        },
+      },
+      missingPackages?.length && {
         title: 'Adding required api packages...',
         task: async () => {
           await execa('yarn', [
             'workspace',
             'api',
             'add',
             '-D',
-            ...providerData.apiPackages,
+            ...missingPackages,
```

Comment:
This code came from the AWS Serverless provider, which made the assumptions 1) packages would be installed in the api/package.json (aka 'api' workspace in Yarn-speak) and 2) they would be 'devDependencies'. 

The package `@redwoodjs/api-server` is now included in the template for all new projects as a regular **dependecies** package (not dev) in api/package.json. The reason I suggested a check is that any Redwood projects created before v0.28.1 might _not_ have api-server installed.

This current code to "check-if-installed" _then_ "install-if-does-not-exist" could cause issues in the following ways:
1. If api-server is already included as a "dependencies" package, it will be redundantly installed as a "devDependencies" package (that's the `-D` option above)
2. If api-server is not included at all, it will be incorrectly installed as a "devDependencies" instead of a "dependencies" package

> Note: if this is starting to feel like a hassle, you could just 1) check for api-server as a "dependencies" package and 2) if not found, provide a warning output with instructions to manually check and add

We can use the const naming `apiPackages` to target the Yarn workspace. If in the future another provider requires installation in root or web workspaces, then they could create `rootPackages` and `webPackages`.

To handle setting a package as either `devDependencies` or `dependencies`, what if you did something like  `export const apiPackages = [{'@redwoodjs/api-server', 'dependencies'}]` to handle the type of package, which could default to `devDependencies`? And although it would be less time-efficient, you could just loop the `yarn workspace api add ...` for each package to install, adding the `-D` option as needed.

Consider all of this me "thinking out loud" and by example. As long as the overall direction is clear, feel free to determine the approach that makes sense to you. (Or, as I mentioned above, skip all this and just check + warn.)

Reply:
I took a bit of a different approach than you suggested but I think reaching the same desired outcome.

There are now two separate consts; `apiPackages` and `apiDevPackages`. The former gets installed without using the `-D` flag.

I updated the `aws-serverless.js` file to use `apiDevPackages` such that I could make use of `apiPackages` for Render.

Let me know what you think.

Reply:
Ok, let's go with it. After seeing the work it took I was a) grateful for your work and b) debated if it was worth it. 

At the very least, because it checks for existing and does not automatically install it will ensure the package versions stay in sync (e.g. if the project is behind a few versions but api-server was unintentionally updated to latest, that could cause a mess of problems). 

Let's do it!